File: app/src/main/java/com/jermaine/awesomelibraries/view/adapter/repolist/viewholder/RepoViewHolder.java
package com.jermaine.awesomelibraries.view.adapter.repolist.viewholder;


import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.jermaine.awesomelibraries.R;

import java.util.List;

import butterknife.BindView;
import butterknife.BindViews;
import butterknife.ButterKnife;

public class RepoViewHolder extends RecyclerView.ViewHolder {
    @BindViews({R.id.id, R.id.name, R.id.description})
    public List<TextView> mTextViews;

    @BindView(R.id.cardView)
    public CardView mCardView;

    public RepoViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }
}

File: app/src/main/java/com/jermaine/awesomelibraries/uicontroller/adapter/repolist/viewholder/RepoViewHolder.kt
package com.jermaine.awesomelibraries.uicontroller.adapter.repolist.viewholder


import android.support.v7.widget.RecyclerView
import android.view.View
import com.jermaine.awesomelibraries.datamodel.repo.Repo
import kotlinx.android.synthetic.main.item_layout.view.*

class RepoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    fun bindRepo(repo: Repo, clickListener: View.OnClickListener) {
        with(repo) {
            itemView.idTextView.text = id
            itemView.nameTextView.text = name
            itemView.descriptionTextView.text = description
            itemView.cardView.setOnClickListener(clickListener)
        }
    }
}